src/Components/Web.JS/src/Rendering/JSRootComponents.ts

```diff
@@ -7,6 +7,12 @@ let nextPendingDynamicRootComponentIdentifier = 0;
 type ComponentParameters = object | null | undefined;
 
 let manager: DotNet.DotNetObject | undefined;
+let initializationCallbacks : { failedToInitialize?: (reason?: any) => void, initialized?: () => void } = {};
+let initializerPromise = new Promise<void>((resolve, reject) => {
+  initializationCallbacks.initialized = resolve;
+  initializationCallbacks.failedToInitialize = reject;
+});
```

Comment:
```suggestion
let initializerResolver: () => void;
const initializerPromise = new Promise<void>((resolve) => initializerResolver = resolve);
```

Then we can also avoid the "if is null" checks later